schema_based_on_diagram.sql

```diff
@@ -0,0 +1,56 @@
+CREATE DATABASE Hospital
+
+-- Create the patients table
+CREATE TABLE patients (
+    id SERIAL PRIMARY KEY,
+    name VARCHAR(100) NOT NULL,
+    date_of_birth DATE NOT NULL
+);
+
+-- Create the medical_histories table
+CREATE TABLE medical_histories (
+    id SERIAL PRIMARY KEY,
+    admitted_at TIMESTAMP NOT NULL,
+    status VARCHAR(50) NOT NULL,
+    patient_id INT NOT NULL,
+    FOREIGN KEY (patient_id) REFERENCES patients(id) ON DELETE CASCADE
+);
+
+-- Create the invoices table
+CREATE TABLE invoices (
+    id SERIAL PRIMARY KEY,
+    total_amount DECIMAL(10, 2) NOT NULL,
+    generated_at TIMESTAMP NOT NULL,
+    payed_at TIMESTAMP,
+    medical_history_id INT NOT NULL,
+    FOREIGN KEY (medical_history_id) REFERENCES medical_histories(id) ON DELETE CASCADE
+);
+
+
+-- Create the treatments table
+CREATE TABLE treatments (
+    id SERIAL PRIMARY KEY,
+    type VARCHAR(50) NOT NULL,
+    name VARCHAR(100) NOT NULL
+);
+
+-- Create the invoice_items table
+CREATE TABLE invoice_items (
+    id SERIAL PRIMARY KEY,
+    unit_price DECIMAL(10, 2) NOT NULL,
+    quantity INT NOT NULL,
+    total_price DECIMAL(10, 2) NOT NULL,
+    invoice_id INT NOT NULL,
+    treatment_id INT NOT NULL,
+    FOREIGN KEY (invoice_id) REFERENCES invoices(id) ON DELETE CASCADE,
+    FOREIGN KEY (treatment_id) REFERENCES treatments(id) ON DELETE CASCADE
+);
+
+-- Create the bridge table for the many-to-many relationship between medical_histories and treatments
+CREATE TABLE treatments_medical_histories (
+    medical_history_id INT NOT NULL,
+    treatment_id INT NOT NULL,
+    PRIMARY KEY (medical_history_id, treatment_id),
+    FOREIGN KEY (medical_history_id) REFERENCES medical_histories(id) ON DELETE CASCADE,
+    FOREIGN KEY (treatment_id) REFERENCES treatments(id) ON DELETE CASCADE
+);
```

Comment:
- Great job for creating the database based on the Given ER diagram, I think it would be  better if you Create FOREIGN KEYS indexes to fit the given [requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/database_from_diagram.md) 👍 
* `CREATE INDEX ON medical_histories (patient_id);`
* `CREATE INDEX ON invoices (medical_history_id);`
* `CREATE INDEX ON invoice_items (invoice_id);`
* `CREATE INDEX ON invoice_items (treatment_id);`
* `CREATE INDEX ON medical_histories_has_treatments (medical_history_id);`
* `CREATE INDEX ON medical_histories_has_treatments (treatment_id);`